code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -125,47 +125,37 @@
 
 /datum/reagent/corgium/on_mob_metabolize(mob/living/L)
 	. = ..()
-	new_corgi = new(get_turf(L))
-	new_corgi.key = L.key
-	new_corgi.name = L.real_name
-	new_corgi.real_name = L.real_name
-	ADD_TRAIT(L, TRAIT_NOBREATH, CORGIUM_TRAIT)
-	//hack - equipt current hat
+	var/obj/shapeshift_holder/H = locate() in L
+	if(H)
+		to_chat(L, "<span class='warning'>You're already corgified!</span>")
+		return
+	ADD_TRAIT(L, TRAIT_MUTE, CORGIUM_TRAIT)
+	new_corgi = new(L.loc)
+	//hat check
 	var/mob/living/carbon/C = L
-	if (istype(C))
+	if(istype(C))
 		var/obj/item/hat = C.get_item_by_slot(ITEM_SLOT_HEAD)
-		if (hat)
+		if(hat?.dog_fashion)
 			new_corgi.place_on_head(hat,null,FALSE)
-	L.forceMove(new_corgi)
+	H = new(new_corgi,src,L)
 
 /datum/reagent/corgium/on_mob_life(mob/living/carbon/M)
 	. = ..()
 	//If our corgi died :(
 	if(new_corgi.stat)
 		holder.remove_all_type(type)
+		addtimer(CALLBACK(src, .proc/restore, M), 2 SECONDS)
```

Comment:
Does on_mob_life get called on the death of the corgi even?

Reply:
The reagent is still inside and processing inside the original mob. This means every ``on_mob_life()`` the stat of the corgi is checked. So when the corgi dies, the original mob is still processing the reagent, that checks the if the corgi is dead, if it is it deletes the corgi, moving the original mob out, and applies the corgi's damage to the mob.
![image](https://user-images.githubusercontent.com/40559528/179176995-0f16164a-6232-4797-a2e9-12691f3d356f.png)